pyuvdata/uvdata/ms.py

```diff
@@ -399,62 +1871,138 @@ def read_ms(
+                # Reach here if no column named SOURCE_ID exists, or if it does exist
+                # but is completely unfilled. Nothing to do at this point but move on.
+                pass
+            # Field names are allowed to be the same in CASA, so if we detect
+            # conflicting names here we use the FIELD row numbers to try and
+            # differentiate between them.
+            field_name_list = tb_field.getcol("NAME")
+            uniq_names, uniq_count = np.unique(field_name_list, return_counts=True)
+            rep_name_list = uniq_names[uniq_count > 1]
+            for rep_name in rep_name_list:
+                rep_count = 0
+                for idx in range(len(field_name_list)):
+                    if field_name_list[idx] == rep_name:
+                        field_name_list[idx] = "%s-%03i" % (
+                            field_name_list[idx],
+                            rep_count,
+                        )
+                        rep_count += 1
+
+            for field_idx in field_list:
+                radec_center = tb_field.getcell("PHASE_DIR", field_idx)[0]
+                field_name = field_name_list[field_idx]
+                if ref_dir_colname is not None:
+                    frame_tuple = self._parse_casa_frame_ref(
+                        ref_dir_dict[tb_field.getcell(ref_dir_colname, field_list[0])]
+                    )
+                else:
+                    frame_tuple = (self.phase_center_frame, self.phase_center_epoch)
+
+                self._add_phase_center(
+                    field_name,
+                    cat_type="sidereal",
+                    cat_lon=radec_center[0],
+                    cat_lat=radec_center[1],
+                    cat_frame=frame_tuple[0],
+                    cat_epoch=frame_tuple[1],
+                    info_source="file",
+                    cat_id=field_id_dict[field_idx],
+                )
+            # Only thing left to do is to update the IDs for the per-BLT records
+            self.phase_center_id_array = np.array(
+                [field_id_dict[sou_id] for sou_id in self.phase_center_id_array],
+                dtype=int,
             )
-        if len(self.nsample_array.shape) == 3:
-            self.nsample_array = np.expand_dims(self.nsample_array, axis=1)
-        self.object_name = tb_field.getcol("NAME")[0]
+
         tb_field.close()
-        tb.close()
 
         if proc is not None:
             proc.join()
         # Fill in the apparent coordinates here
         self._set_app_coords_helper()
 
+        self.data_array = np.expand_dims(self.data_array, 1)
+        self.nsample_array = np.expand_dims(self.nsample_array, 1)
+        self.flag_array = np.expand_dims(self.flag_array, 1)
+        self.freq_array = np.expand_dims(self.freq_array, 0)
+
         # order polarizations
-        self.reorder_pols(order=pol_order)
+        self.reorder_pols(order=pol_order, run_check=False)
+
+        # This is a thing
```

Comment:
What kind of thing?

Reply:
The most thingiest thing that you've ever seen!

Reply:
This got deleted as part of moving the flip check out of `read_ms`.